### PR TITLE
feat(button, button-minor, button-toggle-group): add support for aria-labelledby and aria-describedby (FE-6364)

### LINKED ISSUE #6567: Add `aria-descibedby` support to buttons
### Desired behaviour

When using a screen reader buttons with identical text (in the case of a product catalog) require additional context, such as heading text. Adding support for `aria-describedby` will allow us to add context without causing the accessible name to deviate from the button text.

### Current behaviour

Buttons do not support a `aria-describedby` prop.

### Suggested Solution

_No response_

### CodeSandbox or Storybook URL

_No response_

### Anything else we should know?

_No response_

### Confidentiality

- [X] I confirm there is no confidential or commercially sensitive information included.

### PR DESCRIPTION
**Pull request by Parsium**
fixes #6567

### Proposed behaviour

- Add support for `aria-labelledby` and `aria-describedby` for `Button` and `ButtonMinor`
- Add support for `aria-describedby` for `ButtonToggleGroup` - accessible description is set on each toggle button inside the group ([as recommended by this source](https://devdoc.net/web/developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_group_role.html)).
- Simplify `Button` tests
- Update copyright dates

### Current behaviour

- Accessible description cannot be set on `Button` or `ButtonMinor`.[^1]
- Accessible description cannot be set on `ButtonToggleGroup`
- Copyright dates in README and licence are outdated

### Checklist

- [x] Commits follow our style guide
- [x] Related issues linked in commit messages if required
- [x] Screenshots are included in the PR if useful
- [x] All themes are supported if required
- [x] Unit tests added or updated if required
- [x] Playwright automation tests added or updated if required
- [x] Storybook added or updated if required
- [x] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [x] Typescript `d.ts` file added or updated if required
- [x] Related docs have been updated if required

#### QA

- [x] Tested in provided StackBlitz sandbox/Storybook
- [ ] Add new Playwright test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Testing instructions

Test with the following screen readers: **macOS VoiceOver (with Safari)** and **NVDA (with Chrome and Firefox)**.

- For `Button`:
  1. Open the test story "With External Labels".
  2. With a screen reader active, focus on the button.
  3. Verify if the accessible name "Add Product Addon" and description "You will not be charged until you checkout" are both read out.

- For `ButtonMinor`:
  1. Open the test story "With External Labels".
  2. With a screen reader active, focus on the button.
  3. Verify if the accessible name "Add Product Addon" and description "You will not be charged until you checkout" are both read out.
 
- For `ButtonToggle`:
  1. Open the story "with inputHint".
  2. With a screen reader active, focus on any of the three toggle buttons.
  3. Verify if the accessible name "inputHint example" and description "Hint text" are both read out.

[^1]: Possible to pass any HTML button attribute as a prop, but TypeScript throws an error as they're not officially part of either component's interfaces. https://github.com/Sage/carbon/blob/d0524b57701c790429dfa7e50361897e3841f552/src/components/button/button.component.tsx#L320

### CODE REVIEW

File: src/components/button/button.spec.tsx [hunk 1]
```diff
@@ -1,109 +1,102 @@
...
+      expect(loggerSpy).toHaveBeenCalledTimes(1);
+    });
+  });
+
+  testStyledSystemSpacing((props) => <Button {...props}>Test</Button>, {
+    px: "24px",
+    m: "0",
   });
 
+  describe.each([
```
Comment (edleeks87) [inline src/components/button/button.spec.tsx:81]:
```suggestion
  it.each([
```

File: src/components/button/button.spec.tsx [hunk 2]
```diff
@@ -121,234 +114,148 @@ describe("Button", () => {
...
-      wrapper = render({ children: "foo" }, TestRenderer.create).toJSON();
+    consoleSpy.mockRestore();
+  });
+
+  it.each([
+    [
+      "primary",
+      "var(--colorsActionDisabled500)",
+      "transparent",
+      "var(--colorsActionMajorYin030)",
```
Comment (edleeks87) [inline src/components/button/button.spec.tsx:206]: Suggestion: as every `color` is the same value we could probably simplify this array by removing it and just asserting `color: var(--colorsActionMajorYin030)` in the test

File: src/components/button/button.spec.tsx [hunk 3]
```diff
@@ -460,441 +362,361 @@ describe("Button", () => {
...
-              wrapper
-            );
-          });
-        }
+  it.each([
+    [
+      "primary",
+      "var(--colorsActionDisabled500)",
+      "transparent",
+      "var(--colorsActionMajorYin030)",
```
Comment (edleeks87) [inline src/components/button/button.spec.tsx:405]: Same suggestion as above wrt `color`

File: src/components/button/button.spec.tsx
Comment (edleeks87) [inline src/components/button/button.spec.tsx:1]: praise: good work removing `react-test-renderer` and flattening out some of these tests 👏

File: src/components/button/button.spec.tsx [hunk 4]
```diff
@@ -199,39 +199,13 @@ describe("Button", () => {
...
-      "transparent",
-      "transparent",
-      "var(--colorsActionMajorYin030)",
-    ],
-    [
-      "dashed",
-      "transparent",
-      "var(--colorsActionDisabled500)",
-      "var(--colorsActionMinorYin030)",
-    ],
```
Comment (Parsium) [inline src/components/button/button.spec.tsx]: Decided to remove `dashed` cases, given the variant is now deprecated.

**Review by edleeks87 — DISMISSED** [review]

File: src/components/button-minor/button-minor-test.stories.tsx [hunk 5]
```diff
@@ -10,7 +10,7 @@ import {
 
 export default {
   title: "Button Minor/Test",
-  includeStories: ["DefaultStory"],
+  excludeStories: ["ButtonMinorCustom", "ButtonMinorDifferentTypes"],
```
Comment (tomdavies73) [inline src/components/button-minor/button-minor-test.stories.tsx:13]: **question**: what was the reason for excluding these stories? The `ButtonMinorDifferentTypes` story especially seems useful to keep

Comment (Parsium) [reply src/components/button-minor/button-minor-test.stories.tsx:13]: Good point 👍🏼 These stories appear to be old artifacts for our Cypress/Playwright tests.

Before we migrated all the Cypress tests to TypeScript, we decided to temporarily place the mock components within the `test-stories.tsx` files so we could type-check them. We also have these mock components defined in the `components.test-pw.tsx` file.

If having these mock components as Storybook test stories would be valuable, I'm happy to include them again 😄

**Comment by Parsium** [comment]
Updated changes to `ButtonToggleGroup` after rebasing with `master`. `aria-describedby` is now placed on the group container, rather than passed to each toggle button.

**Review by edleeks87 — APPROVED** [review]

**Review by tomdavies73 — DISMISSED** [review]

**Review by tomdavies73 — DISMISSED** [review]

**Review by edleeks87 — APPROVED** [review]

**Review by divyajindel — APPROVED** [review]

**Pull request merged**